### PR TITLE
[PARQUET-2422] Prevent unwrapping of Hadoop filestreams

### PR DESCRIPTION
**Pull request by rathinb-db**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references
  them in the PR title. For example, "PARQUET-1234: My Parquet PR"
    - https://issues.apache.org/jira/browse/PARQUET-2422
    - In case you are adding a dependency, check if the license complies with
      the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

Refactor Hadoop filestreams to prevent filestream unwrapping (and keeping the original filestream type).
### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Tests not necessary since this is a refactor.

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines
  from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    1. Subject is limited to 50 characters (not including Jira issue reference)
    1. Subject does not end with a period
    1. Subject uses the imperative mood ("add", not "adding")
    1. Body wraps at 72 characters
    1. Body explains "what" and "why", not "how"

### Style
- [x] My contribution adheres to the code style guidelines and Spotless passes.
    - To apply the necessary changes, run `mvn spotless:apply -Pvector-plugins`

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by wgtmac** [comment]
Could you please create a JIRA?

**Review by wgtmac — APPROVED** [review]
It looks trivial to me. Thanks!

It would be good if @steveloughran @Fokko can double check as I'm not that familiar with its history.

**Comment by steveloughran** [comment]
Not sure why there's a need to return a function rather than do the work -is there some other code which needs this?

In an ideal world parquet would be hadoop 3.3+ only and life would be simpler, not just here but with openFile(), ByteBufferPositionedReadable and more

**Comment by wgtmac** [comment]
@shangxinli @gszadovszky @ConeyLiu Do you have any comment?

**Review by gszadovszky — APPROVED** [review]
It looks good to me.

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopStreams.java [hunk 1]
```diff
@@ -77,16 +78,17 @@ public static SeekableInputStream wrap(FSDataInputStream stream) {
    * @param stream stream to probe
    * @return A H2SeekableInputStream to access, or H1SeekableInputStream if the stream is not seekable
```
Comment (ConeyLiu) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopStreams.java:79]: Documentation should also be updated

Comment (wgtmac) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopStreams.java:79]: @rathinb-db Could you resolve this comment? Thanks!

**Pull request merged**